google_guest_agent/main.go

```diff
@@ -262,21 +263,18 @@ func closer(c io.Closer) {
 }
 
 func main() {
-	ctx := context.Background()
-
-	var action string
-	if len(os.Args) < 2 {
-		action = "run"
-	} else {
-		action = os.Args[1]
-	}
-
-	if action == "noservice" {
-		runAgent(ctx)
-		os.Exit(0)
-	}
+	ctx, cancelContext := context.WithCancel(context.Background())
+	sigBus := make(chan os.Signal, 1)
+
+	signal.Notify(sigBus, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
Do we also need to listen any other signals like SIGHUP/SIGABRT?

Reply:
`SIGHUP` doesn't make sense in the context, `SIGABRT` will not be the case as systemd will never do an `abort()` and the signal expects the process to finish ASAP so not much of anything to expect from a handler. `SIGQUIT` however, makes sense as it's a similar signal to SIGTERM but with a slightly different condition. So yep, adding `SIGQUIT`.

Reply:
Done. Thanks.